Utilities/rXml.cpp

```diff
@@ -18,37 +18,50 @@ rXmlNode::rXmlNode(void *ptr)
 	handle = ptr;
 }
 
+// These functions go unused.
+/*
 rXmlNode::rXmlNode(const rXmlNode& other)
 {
 	ownPtr = true;
-	handle = reinterpret_cast<void *>(new wxXmlNode(*reinterpret_cast<wxXmlNode*>(other.handle)));
+	handle = reinterpret_cast<void *>(new pugi::xml_node(*reinterpret_cast<pugi::xml_node*>(other.handle)));
 }
 
 rXmlNode &rXmlNode::operator=(const rXmlNode& other)
 {
 	if (ownPtr)
 	{
-		delete reinterpret_cast<wxXmlNode*>(handle);
+		delete reinterpret_cast<pugi::xml_node*>(handle);
 	}
-	handle = reinterpret_cast<void *>(new wxXmlNode(*reinterpret_cast<wxXmlNode*>(other.handle)));
+	handle = reinterpret_cast<void *>(new pugi::xml_node(*reinterpret_cast<pugi::xml_node*>(other.handle)));
 	ownPtr = true;
 	return *this;
 }
+*/
 
 rXmlNode::~rXmlNode()
 {
 	if (ownPtr)
 	{
-		delete reinterpret_cast<wxXmlNode*>(handle);
+		delete reinterpret_cast<pugi::xml_node*>(handle);
 	}
 }
 
 std::shared_ptr<rXmlNode> rXmlNode::GetChildren()
 {
-	wxXmlNode* result = reinterpret_cast<wxXmlNode*>(handle)->GetChildren();
+	// children() returns an iterator, so we must convert the iterator to a node pointer.
+	pugi::xml_object_range<pugi::xml_node_iterator> it = reinterpret_cast<pugi::xml_node*>(handle)->children();
+	u64 it_size = std::distance(it.begin(), it.end());
+	std::unique_ptr<pugi::xml_node[]> result(new pugi::xml_node[it_size*sizeof(pugi::xml_node)]);
```

Comment:
Pugixml only seems to give back the children as a set of Iterator objects, and it would not allow me to cast or reinterpret the objects as xml_nodes. So I loop over the children to put them in contiguous memory. I will try and see if returning just the "begin()" object will suffice.